src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String MESSAGE = "the index is greater than size";
+    private int size;
+    private T[] arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    private T[] beginning;
+    private T[] end;
+
+    private void resize() {
+        if (size == arrayList.length) {
+            beginning = arrayList;
+            arrayList = (T[]) new Object[(int) (size * 1.5)];
+            for (int i = 0; i < beginning.length; i++) {
+                arrayList[i] = beginning[i];
+            }
+        }
+    }
+
     @Override
     public void add(T value) {
 
+        resize();
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(MESSAGE);
+        }
+        if (size == index) {
+            add(value);
+        } else {
+            resize();
+            for (int i = size; i > index; i--) {
+                arrayList[i] = arrayList[i - 1];
+            }
+            arrayList[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            resize();
+            arrayList[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(MESSAGE);
+        }
```

Comment:
you have the same check in several places, create for this block of code a separate method